src/java/com/reucon/openfire/plugin/archive/xep0313/IQQueryHandler.java

```diff
@@ -265,39 +265,86 @@ private IQ buildForbiddenResponse(IQ packet) {
         String withField = null;
         String startField = null;
         String endField = null;
+        final MonitoringPlugin plugin = (MonitoringPlugin) XMPPServer.getInstance().getPluginManager().getPlugin(MonitoringConstants.NAME);
+        final ConversationManager conversationManager = (ConversationManager)plugin.getModule( ConversationManager.class);
+
+        
         DataForm dataForm = queryRequest.getDataForm();
         if(dataForm != null) {
             if(dataForm.getField("with") != null) {
                 withField = dataForm.getField("with").getFirstValue();
+                Log.debug("MAM: withField="+withField!=null?withField:"NULL");
             }
             if(dataForm.getField("start") != null) {
                 startField = dataForm.getField("start").getFirstValue();
+                Log.debug("MAM: startField="+startField!=null?startField:"NULL");
             }
             if(dataForm.getField("end") != null) {
                 endField = dataForm.getField("end").getFirstValue();
+                Log.debug("MAM: endField="+endField!=null?endField:"NULL");
             }
         }
 
-        Date startDate = null;
-        Date endDate = null;
-        try {
-            if(startField != null) {
-                startDate = xmppDateTimeFormat.parseString(startField);
-            }
-            if(endField != null) {
-                endDate = xmppDateTimeFormat.parseString(endField);
-            }
-        } catch (ParseException e) {
-            Log.error("Error parsing query date filters.", e);
+       try
+        {
+	        long newDate = System.currentTimeMillis()-(conversationManager.getMaxRetrievable() * JiveConstants.DAY);
+	       
+	        Date startDate = null;
+	        Date endDate = null;
+	        try {
+	            if(startField != null) {
+	            	if (conversationManager.getMaxRetrievable()<=0)
+	            	{
+	            		startDate = xmppDateTimeFormat.parseString(startField);
+	            	}
+	            	else
+	            	{
+	            		startDate = new Date(newDate);
```

Comment:
I think that there is a bug here.

If the client provides a "start date" and the admin of Openfire has configured a "max retrievable", then this code will always use the value of "max retrievable". 

Consider this scenario:
- The client wants to retrieve everything from the last hour, and uses 'start date for this.
- The server has a "max retrievable" of 2 weeks.

Using the code in this PR, the client would receive 2 weeks worth of data, instead of one hour.